Index Cutoff: 30
Read Cutoff: 25
If the average of the index of read was below the cutoff, the record was added to the bad file

Total Records: 363246735
zcat /projects/bgmp/shared/2017_sequencing/1294_S1_L008_R1_001.fastq.gz | wc -l
1452986940/4

There will a total of 510351 swapped records that also had high quality (If the indexes or reads had low quality, they were written to the bad file). This is .14% of the total reads. 

```
B1	GTAGCGTA: 7213997/363246735 = 1.99% 
A5	CGATCGAT: 5186748/363246735 = 1.43%
C1	GATCAAGG: 5873900/363246735 = 1.62%
B9	AACAGCGA: 8088051/363246735 = 2.23%   
C9	TAGCCATG: 9618892/363246735 = 2.65% 
C3	CGGTAATC: 4337213/363246735 = 1.19%
B3	CTCTGGAT: 31423030/363246735 = 8.65% 
C4	TACCGGAT: 67805211/363246735 = 18.67%   
A11	CTAGCTCA: 15977593/363246735 = 4.40%
C7	CACTTCAC: 3774585/363246735 = 1.04%
B2	GCTACTCT: 6503447/363246735 = 1.79% 
A1	ACGATCAG: 7367727/363246735 = 2.03%
B7	TATGGCAC: 10054159/363246735 = 2.77% 
A3	TGTTCCGT: 14599903/363246735 = 4.02% 
B4	GTCCTAAG: 8027440/363246735 = 2.21%
A12	TCGACAAG: 3500622/363246735 = 0.96% 
C10	TCTTCGAC: 38173259/363246735 = 10.51% 
A2	ATCATGCG: 9174693/363246735 = 2.53%
C2	ATCGTGGT: 6215981/363246735 = 1.71% 
A10	TCGAGAGT: 10434128/363246735 = 2.87% 
B8	TCGGATTC: 4076058/363246735 = 1.12%
A7	GATCTTGC: 3351378/363246735 = 0.92% 
B10	AGAGTCCA: 10267335/363246735 = 2.83%
A8	AGGATAGC: 7929708/363246735 = 2.18%
Bad Records: 63761326/363246735 = 17.55%
Swapped Records: 510351/363246735 = 0.14%
```

```
Line Counts from Each File:
AACAGCGA_r1.fastq.gz
32352204
AACAGCGA_r2.fastq.gz
32352204
ACGATCAG_r1.fastq.gz
29470908
ACGATCAG_r2.fastq.gz
29470908
AGAGTCCA_r1.fastq.gz
41069340
AGAGTCCA_r2.fastq.gz
41069340
AGGATAGC_r1.fastq.gz
31718832
AGGATAGC_r2.fastq.gz
31718832
ATCATGCG_r1.fastq.gz
36698772
ATCATGCG_r2.fastq.gz
36698772
ATCGTGGT_r1.fastq.gz
24863924
ATCGTGGT_r2.fastq.gz
24863924
bad_read1.fastq.gz
255045304
bad_read2.fastq.gz
255045304
CACTTCAC_r1.fastq.gz
15098340
CACTTCAC_r2.fastq.gz
15098340
CGATCGAT_r1.fastq.gz
20746992
CGATCGAT_r2.fastq.gz
20746992
CGGTAATC_r1.fastq.gz
17348852
CGGTAATC_r2.fastq.gz
17348852
CTAGCTCA_r1.fastq.gz
63910372
CTAGCTCA_r2.fastq.gz
63910372
CTCTGGAT_r1.fastq.gz
125692120
CTCTGGAT_r2.fastq.gz
125692120
GATCAAGG_r1.fastq.gz
23495600
GATCAAGG_r2.fastq.gz
23495600
GATCTTGC_r1.fastq.gz
13405512
GATCTTGC_r2.fastq.gz
13405512
GCTACTCT_r1.fastq.gz
26013788
GCTACTCT_r2.fastq.gz
26013788
GTAGCGTA_r1.fastq.gz
28855988
GTAGCGTA_r2.fastq.gz
28855988
GTCCTAAG_r1.fastq.gz
32109760
GTCCTAAG_r2.fastq.gz
32109760
swapped_read1.fastq.gz
2041404
swapped_read2.fastq.gz
2041404
TACCGGAT_r1.fastq.gz
271220844
TACCGGAT_r2.fastq.gz
271220844
TAGCCATG_r1.fastq.gz
38475568
TAGCCATG_r2.fastq.gz
38475568
TATGGCAC_r1.fastq.gz
40216636
TATGGCAC_r2.fastq.gz
40216636
TCGACAAG_r1.fastq.gz
14002488
TCGACAAG_r2.fastq.gz
14002488
TCGAGAGT_r1.fastq.gz
41736512
TCGAGAGT_r2.fastq.gz
41736512
TCGGATTC_r1.fastq.gz
16304232
TCGGATTC_r2.fastq.gz
16304232
TCTTCGAC_r1.fastq.gz
152693036
TCTTCGAC_r2.fastq.gz
152693036
TGTTCCGT_r1.fastq.gz
58399612
TGTTCCGT_r2.fastq.gz
58399612
```